src/opnsense/mvc/app/controllers/OPNsense/Base/ControllerBase.php

```diff
@@ -204,5 +204,10 @@ public function beforeExecuteRoute($dispatcher)
 
         // append ACL object to view
         $this->view->acl = new \OPNsense\Core\ACL();
+        $this->response->setHeader('Content-Security-Policy', "default-src 'self'; script-src 'self' 'unsafe-inline' 'unsafe-eval'; style-src 'self' 'unsafe-inline' 'unsafe-eval';");
```

Comment:
will only allow resources directly from the firewall and additional embedded in the HTML, style attributes and JavaScript eval.